tests/binary_deserialization.cpp

```diff
@@ -0,0 +1,59 @@
+#include <ozo/binary_deserialization.h>
+
+#include <GUnit/GTest.h>
+
+using namespace ozo;
+using namespace ozo::error;
+
+GTEST("ozo::convert") {
+    auto oid_map = empty_oid_map{};
+
+    SHOULD("[return type mismatch error if oid does not match the type]") {
+        int x;
+        EXPECT_EQ(oid_type_mismatch, convert(TEXTOID, nullptr, 0, oid_map, x));
+    }
+
+    SHOULD("[convert INT2OID to int16_t]") {
+        const int16_t expected = 7;
+        int16_t got;
+
+        const int16_t bytes_storage = detail::convert_to_big_endian(expected);
+        const char* bytes = reinterpret_cast<const char*>(&bytes_storage);
```

Comment:
Why perform data conversions in the test? Just define value as it should be.

Reply:
done